app/views/posts/index.html.erb

```diff
@@ -1 +1,36 @@
-<h1>Listing User Posts</h1>
\ No newline at end of file
+<div class="user">
+  <div class="user-photo">
+    <img src="<%= @user.photo %>" alt="User Photo">
+  </div>
+  <div class="user-details">
+    <%= link_to user_url(@user), class: 'name' do %>
+      <%= @user.name %>
+    <% end %>
+    <p class="posts-no">
+      <%= "Number of posts: #{@user.posts_counter}" %>
+    </p>
+  </div>
+</div>
+<% @posts.each_with_index do |post, index| %>
+     <div class = "post">
+        <%= link_to user_post_url(user_id: @user.id, id: post.id), class: 'user-link' do %>
+          <h3>Post #<%= index + 1 %></h3>
+        <% end %>
+        <p><%= post.text %></p>
+          <div class="count">
+            <p>Comments: <%= post.comments_counter %></p>
+            <p>Likes: <%= post.likes_counter %></p>
+          </div>
+     </div>
+     <% if post.comments_counter > 0 %>
+      <ul class="comments">
+            <% post.comments.each do |comment| %>
+                <li><%= comment.author.name %>: <%= comment.text %></li>
+            <% end %>
```

Comment:
- [ ] Dear @Odongo006, Great job adding some of the changes, but as suggested by the previous reviewer, we need to use the [recent_comments](https://github.com/Odongo006/Rails_blog_App/blob/Views/app/models/post.rb) method that is defined for your posts inside the `Post` controller. This way we will render only the most recent comments.